### PR TITLE
Implementation of Swagger Changes for Bulk API

### PR DESCRIPTION
**Pull request by careys7**
### Description

Draft 

Started on work for bulk-api Swagger implementation (magento-engcom/bulk-api#11).

Extracted the existing `rest` references from `Swagger` into `SwaggerWebapi` module for better modularity.

These changes will allow us to add a `SwaggerWebapiAsync` module to add support for the async swagger UI later. 

Still needs:

- Tests added
- Any changes to Swagger generator class to allow WebapiAsync swagger generation  

### Fixed Issues (if relevant)

1. magento-engcom/bulk-api#11

### Manual testing scenarios

1. Visit example.org/swagger 

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/SwaggerWebapi/composer.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "config": {
+        "sort-packages": true
+    },
+    "require": {
+        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0",
+        "magento/framework": "100.3.*",
+        "magento/module-swagger": "100.3.*"
+    },
+    "type": "magento2-module",
+    "version": "100.3.0-dev",
```
Comment (paliarush) [inline app/code/Magento/SwaggerWebapi/composer.json]: Version should start from 100.0.0

File: app/code/Magento/SwaggerWebapi/composer.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "license": [
+        "OSL-3.0",
+        "AFL-3.0"
+    ],
+    "autoload": {
+        "files": [
+            "registration.php"
+        ],
+        "psr-4": {
+            "Magento\\Swagger\\": ""
```
Comment (paliarush) [inline app/code/Magento/SwaggerWebapi/composer.json]: Should be `SwaggerWebapi`

File: app/code/Magento/SwaggerWebapi/composer.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "magento/module-swagger-webapi",
+    "description": "N/A",
+    "config": {
+        "sort-packages": true
+    },
+    "require": {
+        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0",
+        "magento/framework": "100.3.*",
+        "magento/module-swagger": "100.3.*"
```
Comment (paliarush) [inline app/code/Magento/SwaggerWebapi/composer.json]: `magento/module-webapi` can be added to `suggest` section if not required

File: app/code/Magento/SwaggerWebapi/etc/module.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Magento_SwaggerWebapi" setup_version="0.0.1">
```
Comment (paliarush) [inline app/code/Magento/SwaggerWebapi/etc/module.xml]: Better to start `setup_version` from 2.0.0 for consistency, or 1.0.0 (only few occurrences in other modules)

File: app/code/Magento/Swagger/Block/Index.php [hunk 5]
```diff
@@ -22,11 +25,31 @@ private function getParamStore()
         return $this->getRequest()->getParam('store') ?: 'all';
     }
 
+    /**
+     * @return SchemaTypeInterface
+     */
+    private function getParamSchemaType()
```
[the remark this answers is not in the document]
Comment (careys7) [reply app/code/Magento/Swagger/Block/Index.php]: Ok sure. It was to match `getParamStore` but we need a router anyway

File: app/code/Magento/Swagger/view/frontend/layout/swagger_index_index.xml [hunk 6]
```diff
@@ -45,7 +45,11 @@
         <referenceContainer name="page.wrapper" remove="true"/>
         <referenceBlock name="requirejs-config" remove="true"/>
         <referenceContainer name="root">
-            <block name="swaggerUiContent" class="Magento\Swagger\Block\Index" template="Magento_Swagger::swagger-ui/index.phtml"/>
+            <block name="swaggerUiContent" class="Magento\Swagger\Block\Index" template="Magento_Swagger::swagger-ui/index.phtml">
+                <arguments>
+                    <argument name="schema_types" xsi:type="object">Magento\Swagger\Api\Block\SchemaTypesInterface</argument>
```
[the remark this answers is not in the document]
Comment (careys7) [reply app/code/Magento/Swagger/view/frontend/layout/swagger_index_index.xml]: This was to expose a nice API for other modules but I can remove it

**Comment by careys7** [comment]
Closed, has been updated in #9

**Pull request closed without merging**